### PR TITLE
Adding CI Script to build, test, cover project 

### LINKED ISSUE #207: Setup CI for automatically building and testing changes to source code
A free to use CI service like Travis CI/circle CI should be used to automatically test the changes made to the codebase and run tests. 

This would
- test changes before the merger
- make sure tests are passing before accepting PR
- ease the method to analyze code before the merge.

### PR DESCRIPTION
**Pull request by YashKumarVerma**
Fixes #207

Configuration required by admin:
- since many other coding blocks projects use Travis ci, it means coding blocks have set up integration with Travis
- addition of one environment variable in Travis build config, named ***SECRETS*** which stores the contents of secrets-sample.json, working and live secrets.
- database services requested on CI build
- project built 
- coverage checked (bot not integrated, separate task)

### CODE REVIEW

**Comment by YashKumarVerma** [comment]
@championswimmer  @hereisnaman

**Comment by championswimmer** [comment]
Do not @ people everywhere.  When PRs are made then we already receive notification. The extra @ comments pile up on the inbox and over time and slow us down from getting to things.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+services:
+  - postgresql
+
+before_script:
+  - psql -c 'create database boss;' -U postgres
+
+install:
+  - npm install
+  - rm secrets-sample.json
```
Comment (hereisnaman) [inline .travis.yml:14]: why is this required?

Comment (YashKumarVerma) [reply .travis.yml:14]: because we might be needing working secrets (form oneAuth) and making them public is something we wont like.

So this removes the given secrets-sample and then dumps the data from env variable into the file.

Comment (hereisnaman) [reply .travis.yml:14]: Why do you think a committed file in an open source project will contain private secrets?

**Comment by hereisnaman** [comment]
hint: that is no the correct secrets file. read the Readme.

File: .travis.yml (hunk 1 above)
Comment (YashKumarVerma) [reply .travis.yml:14]: okey I git it, it should be secrets.json

**Comment by hereisnaman** [comment]
Enable this for your fork for testing.

**Comment by YashKumarVerma** [comment]
Requesting someone to carry it forward.
